src/components/RequestPanel/RequestStatus.js

```diff
@@ -0,0 +1,282 @@
+import React, { useMemo } from 'react'
+import { Button, GU, IconCheck, IconCross, Info, useTheme } from '@aragon/ui'
+import RequestStepItem from './RequestStepItem'
+import { useWallet } from '../../providers/Wallet'
+import { getProviderFromUseWalletId } from '../../ethereum-providers'
+import {
+  REQUEST_STATUS_CONFIRMED,
+  REQUEST_STATUS_FAILED,
+  REQUEST_STATUS_PENDING,
+  REQUEST_STATUS_PROCESSED,
+  REQUEST_STATUS_PROCESSING_FAILED,
+} from './request-statuses'
+
+function RequestStatus({
+  allSuccess,
+  lastProcessedAt,
+  maxAttemptsReached,
+  onClosePanel,
+  onNextAttempt,
+  requests,
+  requestStatus,
+}) {
+  const theme = useTheme()
+
+  const multipleRequests = requests.length > 1
+
+  return (
+    <div>
+      <div
+        css={`
+          background: ${theme.feedbackSurface};
+          height: ${44 * GU}px;
+          display: flex;
+          flex-direction: column;
+          justify-content: space-evenly;
+          align-items: center;
+        `}
+      >
+        {multipleRequests ? (
+          <MultiStepRequest
+            lastProcessedAt={lastProcessedAt}
+            requests={requests}
+            requestStatus={requestStatus}
+          />
+        ) : (
+          <SingleRequest request={requests[0]} status={requestStatus[0]} />
+        )}
+      </div>
+      <div
+        css={`
+          margin-top: ${2 * GU}px;
+        `}
+      >
+        <RequestStatusInfo
+          allSuccess={allSuccess}
+          multipleRequests={multipleRequests}
+          maxAttemptsReached={maxAttemptsReached}
+          onClosePanel={onClosePanel}
+          onNextAttempt={onNextAttempt}
+          requests={requests}
+          requestStatus={requestStatus}
+        />
+      </div>
+    </div>
+  )
+}
+
+function MultiStepRequest({ lastProcessedAt, requests, requestStatus }) {
+  return requests.map((request, index) => {
+    const status = requestStatus[index]
+
+    return (
+      <RequestStepItem
+        key={index}
+        lastProcessedAt={lastProcessedAt}
+        status={status}
+        stepNumber={index + 1}
+        request={request}
+      />
+    )
+  })
+}
+
+function SingleRequest({ request, status }) {
```

Comment:
More of a nitpick, but what do you think about reorganizing the code here a bit:

- Moving `MultiStepRequest` as its own component
- Moving `SingleRequest` to a `SingleStepRequest` component
- Putting both of these components into a `RequestSteps` wrapper component + file
- Migrating `RequestStepItem` to be an internal component inside this file

We might able be able to take the opportunity to make a pure function that gives us the label text based on the request given, to dedupe some of the strings.

Reply:
Much cleaner 🥇